include/libisns/meson.build

```
@@ -0,0 +1,27 @@
+# meson build control file for open-isns public include files
+
+inc_conf = configuration_data()
+inc_conf.set('RUNDIR', run_dir)
+
+static_hdr_files = files([
+  'attrs.h',
+  'buffer.h',
+  'isns.h',
+  'isns-proto.h',
+  'message.h',
+  'source.h',
+  'types.h',
+  'util.h'])
+
+template_hdr_name = 'paths.h'
+template_hdr_file = files('paths.h.in')
+
+configure_file(
+  input: template_hdr_file,
+  output: template_hdr_name,
+  configuration: inc_conf,
+  install: true,
+  install_dir: get_option('includedir') / 'libisns')
+
+install_headers(static_hdr_files,
+  install_dir: get_option('includedir') / 'libisns')
```

```suggestion
  subdir: 'libisns')
```

This will automatically install within the includedir.

---

turns out I didn't need subdir, either, since the files have "libisns/" prefixes.

---

I think you do, unless you call the installation from `include/meson.build` so that the actual paths given to the function are stored as e.g. `'libisns/attrs.h'` instead of `'attrs.h'`, *and* use Meson 0.63.0 and `preserve_path: true`.

> Disable stripping child-directories from header files when installing.
> 
> This is equivalent to GNU Automake's nobase option.